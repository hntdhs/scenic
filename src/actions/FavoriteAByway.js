import React, { useState, useContext } from "react";
import { useParams } from "react-router-dom";
import BywayApi from "../api/api";
import LoadingSpinner from "../common/LoadingSpinner";
import UserContext from "../auth/UserContext";
import Alert from "../common/Alert";

function FavoriteAByway({id}) {
    const { byway } = useParams();
    const { currentUser } = useContext(UserContext);

    // const [favorite, setFavorite] = useState(false);
    // might need state for changing status on page from something that can be favorited to something that has been favorited - parkFavorited, setParkFavorited

    const [favorite, setFavorite] = useState();
    const [hasBeenSet, setHasBeenSet] = useState(false);

    // useEffect(() => {  
    //     async function favorite(currentUser, byway) {
    //         await BywayApi.favoriteAByway(byway);
    //         setFavorite(true);

    //     }
    //     favorite(currentUser, byway);
    // }, []);

    async function handleSubmit(evt) {
        await BywayApi.favoriteAByway(currentUser.username, id);
        setFavorite(id);
        setHasBeenSet(true);
        // set state variable to true
    }

//     return (
//         <div>
//             <div>
//                 {/* {setFavorite(id)
//                     ?
//                     <Alert messages={["Byway added to your favorites."]} />
//                     :  */}
//                     {/* try in onSubmit? with a state variable that sets whether or not the byway has been favorited */}
//                     <button onClick={handleSubmit}>
//                         Add this byway to your favorites
//                     </button>
//             </div>
//         </div>
//     )
// }

        return (
            <div>
                {setHasBeenSet
                ? (
                <Alert messages={["Byway added to your favorites."]} />
                 ) : (
                <button onClick={handleSubmit}>
                    Add this byway to your favorites
                </button>
                 )}
            </div>
        )
}

export default FavoriteAByway;