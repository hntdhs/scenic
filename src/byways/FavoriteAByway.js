import React, { useState, useContext } from "react";
import { useParams } from "react-router-dom";
import BywayApi from "../api/api";
import LoadingSpinner from "../common/LoadingSpinner";
import UserContext from "../auth/UserContext";

function FavoriteAByway({id}) {
    const { byway } = useParams();
    const { currentUser } = useContext(UserContext);

    // const [favorite, setFavorite] = useState(false);
    // might need state for changing status on page from something that can be favorited to something that has been favorited - parkFavorited, setParkFavorited

    const [favorite, setFavorite] = useState();

    // useEffect(() => {  
    //     async function favorite(currentUser, byway) {
    //         await BywayApi.favoriteAByway(byway);
    //         setFavorite(true);

    //     }
    //     favorite(currentUser, byway);
    // }, []);

    async function handleSubmit(evt) {
        await BywayApi.favoriteAByway(currentUser.username, id);
        setFavorite(id);
    }

    return (
        <div>
            <button onClick={handleSubmit}>
                Add this byway to your favorites
            </button>
        </div>
    )
}

export default FavoriteAByway;