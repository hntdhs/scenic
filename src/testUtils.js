import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};
// const
let UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
    )


// const UserProvider =
//     ({ children, currentUser = demoUser, hasAppliedToJob = () => false }) => (
//     <UserContext.Provider value={{ currentUser, hasAppliedToJob }}>
//       {children}
//     </UserContext.Provider>
// );

export { UserProvider };
