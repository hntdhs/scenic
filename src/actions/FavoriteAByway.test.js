import React from "react";
import { render } from "@testing-library/react";
import FavoriteAByway from "./FavoriteAByway.js"
import { MemoryRouter } from "react-router-dom";
import { UserProvider } from "../testUtils";

it("matches snapshot", function () {

    const { asFragment } = render( 
    <MemoryRouter>
      <UserProvider>
      <FavoriteAByway />
      </UserProvider>
    </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });