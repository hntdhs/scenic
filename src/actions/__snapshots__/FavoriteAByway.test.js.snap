// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Add this byway to your favorites
    </button>
  </div>
</DocumentFragment>
`;
